### PR TITLE
add erlang & elixir example processing applications

### PR DESCRIPTION
**Pull request by zerth**
Resolve examples portion of https://github.com/AdRoll/exmld/issues/4.

### CODE REVIEW

**Review by polvorin — APPROVED** [review]
❤️

File: examples/elixir_processor/lib/elixir_processor.ex [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    state
+  end
+
+  defp flow_add_record(%__MODULE__{pending_items: pending} = state, item) do
+    %{state | pending_items: [item | pending]}
+  end
+
+  # possibly process the current pending batch of records if of the appropriate size or
+  # enough time has elapsed:
+  defp maybe_flush(state) do
```
Comment (marcelog) [inline examples/elixir_processor/lib/elixir_processor.ex:137]: Might be worth to use [Flow.Window.trigger* family of functions](https://hexdocs.pm/flow/Flow.Window.html#trigger/3), or at least mention these here (which seems like the way to do this kind of things).

**Review by marcelog — APPROVED** [review]
This is awesome :) Thinking that in the long term, these can be used both as an example AND a template for new applications, so maybe as possible future improvements:

* Options can be moved to the corresponding config files.
* Some unit tests can be added (so if erlmld/exmld change we can still know if the examples are still "good to go"). These could be run with the main unit tests when we have some.
* Flow.Window could be used.

File: examples/elixir_processor/lib/elixir_processor.ex (hunk 1 above)
Comment (zerth) [reply examples/elixir_processor/lib/elixir_processor.ex:137]: Good idea, will look into it (and also making the exmld user interface more convenient for doing that, e.g., via options to `Exmld.flow/6` or `Exmld.from_stages/1`).

Comment (zerth) [reply examples/elixir_processor/lib/elixir_processor.ex:137]: @marcelog: Added use of a time-based window for the elixir example (& added a new option accepted by `Exmld.flow/6` supporting the new example).

**Comment by zerth** [comment]
Re: moving options to config files; I think that would be worthwhile as long as the resulting example applications continued supporting multiple stream consumers in the same node.  And tests & converting the examples to templates (perhaps with a mix task for creating) also sound good as future improvements.

**Pull request merged**